src/ddspdrum/torchmodule.py

```diff
@@ -485,3 +485,154 @@ def _forward(self, control_in: T, audio_in: T) -> T:
 
         audio_in = fix_length(audio_in, len(control_in))
         return control_in * audio_in
+
+
+class TorchFIR(TorchSynthModule):
+    """
+    A finite impulse response low-pass filter. Uses convolution with a windowed
+    sinc function.
+
+    Parameters
+    ----------
+
+    cutoff (float)      :   cutoff frequency of low-pass in Hz, must be between 5 and
+                            half the sampling rate. Defaults to 1000Hz.
+    filter_length (int) :   The length of the filter in samples. A longer filter will
+                            result in a steeper filter cutoff. Should be greater than 4.
+                            Defaults to 512 samples.
+    sample_rate (int)   :   Sampling rate to run processing at.
+    """
+
+    def __init__(
+        self,
+        cutoff: float = 1000.0,
+        filter_length: int = 512,
+        sample_rate: int = SAMPLE_RATE,
+    ):
+        super().__init__(sample_rate=sample_rate)
+        self.add_parameters(
+            [
+                TorchParameter(
+                    value=cutoff,
+                    parameter_name="cutoff",
+                    parameter_range=ParameterRange(5.0, sample_rate / 2.0, "log")
+                ),
+                TorchParameter(
+                    value=filter_length,
+                    parameter_name="length",
+                    parameter_range=ParameterRange(4.0, 4096.)
+                )
+            ]
+        )
+
+    def _forward(self, audio_in: T) -> T:
+        """
+        Filter audio samples
+        TODO: Cutoff frequency modulation, if there is an efficient way to do it
+
+        Parameters
+        ----------
+
+        audio (T)  :   audio samples to filter
+        """
+
+        impulse = self.windowed_sinc(self.p("cutoff"), self.p("length"))
+        impulse = torch.reshape(impulse, (1, 1, impulse.size()[0]))
```

Comment:
reshape looses gradients I think? You probably want view?

Are you able to backprop through this?

Reply:
Yes able to backprop through this -- is view better though?

Reply:
So I guess reshape will change the view if possible, otherwise it may copy the data -- so I think view is better here, made this change.